include/irr/asset/ICPUComputePipeline.h

```diff
@@ -32,9 +32,9 @@ class ICPUComputePipeline : public IComputePipeline<ICPUSpecializedShader, ICPUP
 		}
 	}
 
-    bool restore(IAsset* _other) override
+    bool canBeRestoredFrom(const IAsset* _other) const override
     {
-        if (!IAsset::restore(_other))
+        if (!IAsset::canBeRestoredFrom(_other))
             return false;
 
```

Comment:
the "does `this` match `_other`"restoration check needs to be fully recursive, so need to check whether layouts match (etc.) before trying to restore top level objects

Reply:
kinda paired up with https://github.com/buildaworldnet/IrrlichtBAW/pull/549#discussion_r507659230